### PR TITLE
adds documentation for sharepoint lists

### PR DESCRIPTION
**Pull request by HiteshRepo**
adds documentation for sharepoint lists.
should go in after: https://github.com/alcionai/corso/pull/5048

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [x] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)
#4754 

#### Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/cli/backup/sharepoint.go [hunk 1]
```diff
@@ -57,7 +61,26 @@ corso backup details sharepoint --backup 1234abcd-12ab-cd34-56de-1234abcd \
...
+corso backup details sharepoint --backup 1234abcd-12ab-cd34-56de-1234abcd \
+    --list-created-after 2024-01-01T12:23:34
+
+# Explore lists created before a given time
+corso backup details sharepoint --backup 1234abcd-12ab-cd34-56de-1234abcd \
+    --list-created-before 2024-01-01T12:23:34
+
+# Explore lists modified before a given time
+corso backup details sharepoint --backup 1234abcd-12ab-cd34-56de-1234abcd \
+    --list-modified-before 2024-01-01T12:23:34
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:79]: something to change in a different PR: I just realized that `list-modified-before` can be interpreted as "had any modification before that time".  I think we mean to say `list-last-modified-before` as in, "the most recent modification was before a certain time", right?

File: website/docs/support/known-issues.md [hunk 2]
```diff
@@ -43,3 +43,7 @@ Below is a list of known Corso issues and limitations:
   updated the locale since the backup was created.
 
 * In-place Exchange contacts restore will merge items in folders named "Contacts" or "contacts" into the default folder.
+
+* Sharepoint list item "attachments" aren't Backup-ed up, Restore-d and Export-ed.
```
Comment (ryanfkeepers) [inline website/docs/support/known-issues.md:47]: no hyphens here, please.
```suggestion
* Sharepoint list item "attachments" aren't backed up, restored or exported.
```

File: website/docs/support/known-issues.md [hunk 3]
```diff
@@ -43,3 +43,7 @@ Below is a list of known Corso issues and limitations:
...
+  This is because of limitation of GRAPH APIs to provide the relevant info in responses.
+  An issue has been created to track the progress: https://github.com/alcionai/corso/issues/5085
```
Comment (ryanfkeepers) [inline website/docs/support/known-issues.md:49]:
```suggestion
  Graph API does not currently provide attachment data for Lists or List Items.
  An issue has been created to track the progress: https://github.com/alcionai/corso/issues/5085
```

**Review by ryanfkeepers — APPROVED** [review]
Approving to unblock.  Also pulling in @gmatev to get a thumbs up on the docs content.

File: website/docs/support/bugs-and-features.md [hunk 4]
```diff
@@ -45,3 +45,151 @@ Next, file a [GitHub issue](https://github.com/alcionai/corso/issues/) with the
 ([default log file](../../setup/configuration/#log-files) and `corso-gc.log`, the Corso GC log file, from above) and
 information on the size of the Exchange mailbox, OneDrive location, or SharePoint site that you are having an issue
 with.
+
+## Sharepoint List Anomalies
+
+Here is an example of GRPAH API response of columns of a list:
```
Comment (pandeyabs) [inline website/docs/support/bugs-and-features.md:51]:
```suggestion
Here is an example of graph API response of columns of a list:
```

File: website/docs/support/known-issues.md [hunk 5]
```diff
@@ -43,3 +43,7 @@ Below is a list of known Corso issues and limitations:
   updated the locale since the backup was created.
 
 * In-place Exchange contacts restore will merge items in folders named "Contacts" or "contacts" into the default folder.
+
+* Sharepoint list item "attachments" aren't backed up, restored or exported.
+  Graph API doesn't currently provide attachment data for Lists or List Items.
+  An [issue](https://github.com/alcionai/corso/issues/5085) has been created to track the progress..
```
Comment (pandeyabs) [inline website/docs/support/known-issues.md:49]: We don't need to mention particular issues here, as those are for internal tracking.

```suggestion
```

File: website/docs/support/bugs-and-features.md [hunk 6]
```diff
@@ -45,3 +45,151 @@ Next, file a [GitHub issue](https://github.com/alcionai/corso/issues/) with the
...
+
+```json
```
Comment (pandeyabs) [inline website/docs/support/bugs-and-features.md:53]: This seems quite verbose IMO. Can we move details to a ticket and link the issue here?

**Pull request merged**